Spoiler flags

change1331 opened the pull request:
Adds flags to the spoiler map. Includes options cog to toggle items and flags on map, these settings are saved. Overhauls some backend for visualizer script. Adds couple items to spoiler.json.

kjbranch commented:
This is starting to look really nice and is getting closer. Some things I would recommend changing ...

On the left panel:
- The `collectible on this step` icons get larger with a mouseover; I think flags should as well.
- The flags for `collectible on this step` can be clicked on to show a route to them. I wonder if it would make sense to allow `previously collectible` to as well - but items don't do that and ammo cant really do that, so I could go either way on it.
- Some of the flags should be removed: f_AllItemsSpawn, f_KilledZebetites, f_MotherBrainGlassBroken, f_TourianOpen, f_AcidChozoWithoutSpaceJump
  - f_DefeatedBombTorizo can probably be removed as well, as it's basically free and that room is cluttered
- The flags are listed before the items in `previously collecible` and `collectible on this step`. I would put the items first.
- The flag icons are too small to see what they are depicting. I would increase their size by 1.5-2x (only on the left panel)
- There is a bit of padding space between the item icons, I think that should be also added between the flags.

The cog dropdown: 
- The cog is hard to see. Maddo said there is a bootstrap icon that is being used on the generate page now that should be used here as well. 
- I'm not sure I'm seeing a benefit to split out the items into different categories. If that remains separate, I think beams should be combined with majors, health should be named `energy` or `tanks` and I don't love that `ammo` doesnt include missiles, but I understand why (it could possibly be named `super, PB`, but idk.
- In Map Rando, we have defined 4 tier where major>tank>super/pb>missiles, so energy/tanks should be moved up above supers/pb
- Sometimes the flag checkboxes didn't work properly, but I couldn't recreate it, so I'm not sure why. Very possible it was user error. 

The Map:
  - f_ZebesAwake icons can be removed from the map. They look to be indicating where that flag can be used, which isn't useful and not really how any of the other flags work.
  - The starting icon, with a ship start, looks like it's starting on the top left ledge - I would center it above the ship.
  - The ship also needs to be moved up a little.
  - I'm thinking that it might be nice to use simple flag/boss icons for the map and more detailed images on the left panel. This could reduce the clutter feel and make it more simple.
    - For instance, the bosses (and maybe minibosses) could use the vanilla icon (it's a little 8x8 ish pixel of a black skull or something). 
    - For all of the other flags, we could use a small flag icon (I love the icon that is used throughout in Celeste Ch7).
    - If this looks good, then the other icons could use some downsizing as well: Start, ship, MB. The ship has an icon on the vanilla map, which would match; I'm not sure on the other two at the moment though

Icon graphics:
- I think some of these can be cropped differently still. Currently all items are 16x16, but the flags vary a lot in shape and resolution. It looks like a challenge to get them to look more similar while remaining recognizable - I was thinking I could take an attempt at it?

change1331 commented:
> * The `collectible on this step` icons get larger with a mouseover; I think flags should as well.
> * The flag icons are too small to see what they are depicting. I would increase their size by 1.5-2x (only on the left panel)
> * There is a bit of padding space between the item icons, I think that should be also added between the flags.
> * The flags are listed before the items in `previously collecible` and `collectible on this step`. I would put the items first.
Ya side bar needs little work I think


> * The flags for `collectible on this step` can be clicked on to show a route to them. I wonder if it would make sense to allow `previously collectible` to as well - but items don't do that and ammo cant really do that, so I could go either way on it.
Probably easy enough to add items are probably bit harder to sort as they can duplicate.

> * Some of the flags should be removed: f_AllItemsSpawn, f_KilledZebetites, f_MotherBrainGlassBroken, f_TourianOpen, f_AcidChozoWithoutSpaceJump
Agreed on mother brain glass/zebs they redundant maybe add tooltips for the others if they are on in the current map rando as they are options?  I'll have to do some testing on the options as I don't even know if they drop into starting flags if they are off.

>   * f_DefeatedBombTorizo can probably be removed as well, as it's basically free and that room is cluttered
Agreed but he can be an objective though so hard to remove him.  Think the minor flags on map will help with this

> The cog dropdown:
> * The cog is hard to see. Maddo said there is a bootstrap icon that is being used on the generate page now that should be used here as well.
Will change

> * I'm not sure I'm seeing a benefit to split out the items into different categories. If that remains separate, I think beams should be combined with majors, health should be named `energy` or `tanks` and I don't love that `ammo` doesnt include missiles, but I understand why (it could possibly be named `super, PB`, but idk.
> * In Map Rando, we have defined 4 tier where major>tank>super/pb>missiles, so energy/tanks should be moved up above supers/pb
ya I wasn't sure whether or not to include beams with majors or not but probably keep it consistent with in game


> * f_ZebesAwake icons can be removed from the map. They look to be indicating where that flag can be used, which isn't useful and not really how any of the other flags work.
zebes awake is weird its clickable on all the rooms its used just kept it consistent with how the rooms worked as is

> * The starting icon, with a ship start, looks like it's starting on the top left ledge - I would center it above the ship.
I think this messed up when I switched to using offsets from json will need to adjust

> * The ship also needs to be moved up a little.


> * Sometimes the flag checkboxes didn't work properly, but I couldn't recreate it, so I'm not sure why. Very possible it was user error.
flags appear under multiple options so if one is on they don't disappear .ie bosses and objectives overlap often.  didn't really know how to handle this very well....

> * I'm thinking that it might be nice to use simple flag/boss icons for the map and more detailed images on the left panel. This could reduce the clutter feel and make it more simple.
>   * For instance, the bosses (and maybe minibosses) could use the vanilla icon (it's a little 8x8 ish pixel of a black skull or something).
>   * For all of the other flags, we could use a small flag icon (I love the icon that is used throughout in Celeste Ch7).
>   * If this looks good, then the other icons could use some downsizing as well: Start, ship, MB. The ship has an icon on the vanilla map, which would match; I'm not sure on the other two at the moment though

...but this could be a good answer.  Objective icon, boss icon, miniboss icon, generic flag. detailed icons on sidebar. would fix the issue of having objective and boss overlap as they would be different icons... 

> Icon graphics:
> * I think some of these can be cropped differently still. Currently all items are 16x16, but the flags vary a lot in shape and resolution. It looks like a challenge to get them to look more similar while remaining recognizable - I was thinking I could take an attempt at it?
Graphics are not my best but with all the directions definitely getting closer and any help is appreciated. Going to try Details in sidebar generics on map next.  I think it will help with clutter a lot definitely a lot of info.

blkerby commented:
> > * f_ZebesAwake icons can be removed from the map. They look to be indicating where that flag can be used, which isn't useful and not really how any of the other flags work.
> 
> zebes awake is weird its clickable on all the rooms its used just kept it consistent with how the rooms worked as is

Yeah, in the beginning I included this because I wanted there to be some way for the player to be able to see the logic for waking Zebes. But I agree it's bad and should be removed. Since flag links were included in the spoiler route, this gives the player a better way to find this information.
